include/oneapi/mkl/dft/detail/descriptor_impl.hpp

```diff
@@ -0,0 +1,97 @@
+#ifndef _ONEMKL_DFT_DESCRIPTOR_IMPL_HPP_
+#define _ONEMKL_DFT_DESCRIPTOR_IMPL_HPP_
+
+#include <cstdint>
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/mkl/types.hpp"
+
+#include "oneapi/mkl/detail/export.hpp"
+#include "oneapi/mkl/detail/get_device_id.hpp"
+#include "oneapi/mkl/dft/types.hpp"
+
+namespace oneapi {
+namespace mkl {
+namespace dft {
+namespace detail {
+
+class descriptor_impl {
```

Comment:
Everything in this impl class is general (not backend specific). So, it does not need to be in descriptor_impl, but can be in the descriptor class. We only need pimpl for backend-specific parts (factorizations, twiddle tables etc) and these can only be initialized in the commit call, not in descriptor constructor.

Reply:
Hey Tadej, thank you for the review, I have a few options to move forward
1. impliment the constructor and the `set_value` +` get_value` in the `descriptor.hpp`, use pimpl only for commit
2. if we prefer to separate implimentation and declaration then I would have to explicitly specialize consturctors + pre-commit member functinos, which would bloat the implimentation code (descriptor.cxx)
3. could use detail::create_descriptor (pointer to class impl) for general implimentation, and detail::create_commit (pointer to different commit class ?) along with mklcpu::creat_commit (... mklgpu::create_commit), this would avoid some bloating as create_descriptor can be templated and passed on to the impl class.

for now I can work on option 1 as the prefered way, let me know your thoughts on this.

Reply:
Yeah, I like 1 the best as well. Other options sound like unnecessary complications.

Reply:
710a4e157c8d3fbbfe2144708e4886066084c4c0, should enable you to work on gpu commit.